checks/microbenchmarks/gpu/node_burn/node-burn.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+import uenv
+
+
+ref_nb_gflops = {
+    'a100': {'nb_gflops': (9746*2*0.85, -0.1, None, 'GFlops')},
+    # https://www.nvidia.com/content/dam/en-zz/Solutions/Data-Center/a100/pdf/
+    # + nvidia-a100-datasheet-us-nvidia-1758950-r4-web.pdf (FP64 Tensor Core)
+    'gh200': {'nb_gflops': (42700, -0.1, None, 'GFlops')},
+    # from https://confluence.cscs.ch/display/SCISWDEV/Feeds+and+Speeds
+}
+
+
+@rfm.simple_test
+class baremetal_cuda_node_burn(rfm.RegressionTest):
+    valid_systems = ['+nvgpu +remote']
+    valid_prog_environs = ['builtin']
+    num_gpus = variable(int, value=4)
+    num_tasks_per_node = 4
+    nb_duration = variable(int, value=10)
+    nb_matrix_size = variable(int, value=30000)
+    # NOTE: nb_matrix_size = parameter([nn for nn in range(4000, 32000, 2000)])
+    executable = 'build/burn'
+    jfrog = variable(
+        str, value='https://jfrog.svc.cscs.ch/artifactory/cscs-reframe-tests')
+    build_system = 'CMake'
+
+    def fullpath(self, files):
+        return next((file for file in files if os.path.exists(file)), None)
+
+    @run_after('init')
+    def set_num_tasks(self):
+        self.num_tasks = self.num_gpus
+        self.extra_resources = {
+            'gres': {'gres': f'gpu:{self.num_tasks_per_node}'}
+        }
+
+    @run_before('compile')
+    def set_compilation_env(self):
+        self.skip_if_no_procinfo()
+
+        gcc_l = [f'/usr/bin/gcc-{gnu_version}' for gnu_version in (13, 12, 11)]
+        self.build_system.cc = self.fullpath(gcc_l)
+        self.build_system.cxx = self.fullpath(gcc_l).replace('gcc', 'g++')
+
+        platform = self.current_partition.processor.platform
+        nvcc_l = [f'/opt/nvidia/hpc_sdk/Linux_{platform}/{year}/cuda/bin/nvcc'
+                  for year in (2025, 2024, 2023)]
+        self.build_system.nvcc = os.path.realpath(self.fullpath(nvcc_l))
+
+        self.prebuild_cmds += [
```

Comment:
Why do we need to download cmake?

Reply:
the version from sles/15 is cmake/3.20.4 which is incompatible with the code